main.tf

```diff
@@ -3,13 +3,16 @@
-  watson_discovery_id            = var.existing_watson_discovery_instance_crn != null ? data.ibm_resource_instance.existing_watson_discovery_instance[0].id : ibm_resource_instance.watson_discovery_instance[0].id
-  watson_discovery_crn           = var.existing_watson_discovery_instance_crn != null ? data.ibm_resource_instance.existing_watson_discovery_instance[0].crn : ibm_resource_instance.watson_discovery_instance[0].crn
-  watson_discovery_guid          = var.existing_watson_discovery_instance_crn != null ? data.ibm_resource_instance.existing_watson_discovery_instance[0].guid : ibm_resource_instance.watson_discovery_instance[0].guid
-  watson_discovery_name          = var.existing_watson_discovery_instance_crn != null ? data.ibm_resource_instance.existing_watson_discovery_instance[0].resource_name : ibm_resource_instance.watson_discovery_instance[0].resource_name
-  watson_discovery_plan_id       = var.existing_watson_discovery_instance_crn != null ? null : ibm_resource_instance.watson_discovery_instance[0].resource_plan_id
-  watson_discovery_dashboard_url = var.existing_watson_discovery_instance_crn != null ? null : ibm_resource_instance.watson_discovery_instance[0].dashboard_url
+  account_id               = var.existing_watson_discovery_instance_crn != null ? module.crn_parser[0].account_id : ibm_resource_instance.watson_discovery_instance[0].account_id
+  watson_discovery_id      = var.existing_watson_discovery_instance_crn != null ? data.ibm_resource_instance.existing_watson_discovery_instance[0].id : ibm_resource_instance.watson_discovery_instance[0].id
+  watson_discovery_crn     = var.existing_watson_discovery_instance_crn != null ? data.ibm_resource_instance.existing_watson_discovery_instance[0].crn : ibm_resource_instance.watson_discovery_instance[0].crn
+  watson_discovery_guid    = var.existing_watson_discovery_instance_crn != null ? data.ibm_resource_instance.existing_watson_discovery_instance[0].guid : ibm_resource_instance.watson_discovery_instance[0].guid
+  watson_discovery_name    = var.existing_watson_discovery_instance_crn != null ? data.ibm_resource_instance.existing_watson_discovery_instance[0].resource_name : ibm_resource_instance.watson_discovery_instance[0].resource_name
+  watson_discovery_plan_id = var.existing_watson_discovery_instance_crn != null ? null : ibm_resource_instance.watson_discovery_instance[0].resource_plan_id
+
+  # Below is the temporary workaround to get the dashboard_url and should be removed later. Refer [Issue-12944](https://github.ibm.com/GoldenEye/issues/issues/12944)
+  raw_crn                        = "crn:v1:bluemix:public:discovery:${var.region}:a/${local.account_id}:${local.watson_discovery_guid}::"
+  watson_discovery_dashboard_url = var.existing_watson_discovery_instance_crn != null ? null : "https://cloud.ibm.com/services/discovery/${urlencode(local.raw_crn)}"
```

Comment:
I think...`url` will be formed in both cases using your logic whether existing_instance_crn is provided or newly created. So `var.existing_watson_discovery_instance_crn != null ? null` may not be required.

Reply:
Right, this seems to be working. 
I removed the condition and checked, so this is redirecting correctly to existing instance. 

Not sure if this will work when workaround will be removed. We can check this later once we receive URL from `watson discovery`. For now this change can be made.

Reply:
Added the change. Marking conversation as resolved.